### ISSUE TITLE
[MNGSITE-547] Broken link in guide about dependency mechanism

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-site#1129]
**[Matthias Bünger](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER302100)** opened **[MNGSITE-547](https://issues.apache.org/jira/browse/MNGSITE-547?redirect=false)** and commented

As of today (2024-12-20) there  is a broken link in the [Guide: Introduction to the Dependency Mechanism](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html)

The TOC contains a link to the BOM section

```
<a href="#bill-of-materials-bom-poms">Bill of Materials (BOM) POMs</a>

leads to

https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#bill-of-materials-bom-poms
```

but the anchor is broken and therefore the link above does not work

```
<a id="Bill_of_Materials_.28BOM.29_POMs">
```

---

**Issue Links:**
- [MNGSITE-513](https://issues.apache.org/jira/browse/MNGSITE-513) Intra-page anchor links from some tables of contents (ToCs) are broken

**Remote Links:**
- [GitHub Pull Request #600
](https://github.com/apache/maven-site/pull/600)

**Comment by jira-importer** [comment]
**[Matthias Bünger](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER302100)** commented

I think the problematic thing here is the automatic transition of the braces.

The source code looks fine .

```
** {Bill of Materials (BOM) POMs}
```

According to the docs they should be stripped, but it looks that Doxia tries to transform them

> Anchor construction
> Contrary to the original APT format, an anchor/link is not its text with all non alphanumeric characters stripped. Ideally, an anchor should be a valid Doxia id, ie it must begin with a letter ([A-Za-z]) and may be followed by any number of letters, digits ([0-9]), hyphens ("-"), underscores ("_"), colons (":"), and periods ("."). Any anchor that does not satisfy this pattern is transformed according to the following rules:
> 
> * Any whitespace at the start and end is removed
> * If the first character is not a letter, prepend the letter 'a'
> * Any spaces are replaced with an underscore '_'
> * Any characters not matching the above pattern are stripped.
> 
> Note in particular that case is preserved in this conversation and that APT anchors and links are case-sensitive. So the anchor for the section title in the previous example would be Anchors_for_section_titles.

So maybe this is an Doxia issue? [michaelo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michaelo) [sjaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sjaranowski)

P.S. Sadly it's only possible to provide a different link/link-text for links, but not for anchors (according to docs)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I bet this documentation is not uptodate because the Doxia ID generation has been improved in the last major version: https://maven.apache.org/doxia/components/doxia-archives/doxia-2.0.0/doxia-core/xref/org/apache/maven/doxia/util/DoxiaUtils.html#L113

**Comment by jira-importer** [comment]
**[Matthias Bünger](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER302100)** commented

> I bet this documentation is not uptodate because the Doxia ID generation has been improved in the last major version: https://maven.apache.org/doxia/components/doxia-archives/doxia-2.0.0/doxia-core/xref/org/apache/maven/doxia/util/DoxiaUtils.html#L113

The Doxia docs are  last published on 2024-07-04, the last Doxia release (2.0.0M12 in May 24 - according to JIRA).
https://maven.apache.org/doxia/references/doxia-apt.html

Aside from this I found that https://maven.apache.org/guides/introduction/introduction-to-profiles.html also has a TOC entry with special chars - and this works - but here the TOC is fully generated, while on the dependency mechanism guide it's manually written.

Edit: PR created

**Comment by jira-importer** [comment]
**[Matthias Bünger](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER302100)** commented

As a note: I'm also working on a bigger PR (in parallel) to fix all toc and heading generations. But this should not provide this issue to be closed after PR is merged.
